src/main/java/org/springframework/integration/kafka/inbound/KafkaInboundGateway.java

```diff
@@ -133,6 +150,45 @@ public void setRecoveryCallback(RecoveryCallback<? extends Object> recoveryCallb
 		this.recoveryCallback = recoveryCallback;
 	}
 
+	/**
+	 * Specify a {@link BiConsumer} for seeks management during
+	 * {@link ConsumerSeekAware.ConsumerSeekCallback#onPartitionsAssigned(Map, ConsumerSeekAware.ConsumerSeekCallback)}
+	 * call from the {@link org.springframework.kafka.listener.KafkaMessageListenerContainer}.
+	 * This is called from the internal {@link RecordMessagingMessageListenerAdapter} implementation.
+	 * @param onPartitionsAssignedCallback the {@link BiConsumer} to use
+	 * @since 3.0.4
+	 * @see ConsumerSeekAware
+	 */
+	public void setOnPartitionsAssignedSeekCallback(
+			BiConsumer<Map<TopicPartition, Long>, ConsumerSeekAware.ConsumerSeekCallback> onPartitionsAssignedCallback) {
+		this.onPartitionsAssignedSeekCallback = onPartitionsAssignedCallback;
+	}
+
+	/**
+	 * Specify a {@link BiConsumer} for seeks management during
+	 * {@link ConsumerSeekAware.ConsumerSeekCallback#onIdleContainer(Map, ConsumerSeekAware.ConsumerSeekCallback)}
+	 * call from the {@link org.springframework.kafka.listener.KafkaMessageListenerContainer}.
+	 * This is called from the internal {@link RecordMessagingMessageListenerAdapter} implementation.
+	 * @param onIdleSeekCallback the {@link BiConsumer} to use
+	 * @since 3.0.4
+	 * @see ConsumerSeekAware
+	 */
+	public void setOnIdleSeekCallback(
+			BiConsumer<Map<TopicPartition, Long>, ConsumerSeekAware.ConsumerSeekCallback> onIdleSeekCallback) {
+		this.onIdleSeekCallback = onIdleSeekCallback;
+	}
+
+	/**
+	 * Set a @code boolean} flag to indicate that request message should have extra headers.
+	 * @param setAdditionalHeaders {@code boolean} flag to add or not extra headers into the message to send.
+	 * @since 3.0.4
+	 * @see IntegrationKafkaHeaders
+	 */
+	public void setAdditionalRequestHeaders(boolean setAdditionalHeaders) {
```

Comment:
This name seems too general; it is specifically related to adding the seek callback to the headers.

Reply:
Yes, for now. I'm open for better name.
Or do you think it would be better to add a `boolean` setter for any new headers which may come in the future?